HW2/P2/parallel_vector.pyx

```diff
@@ -75,16 +75,23 @@ cpdef move_data_fine_grained(np.int32_t[:] counts,
        omp_lock_t *locks = get_N_locks(counts.shape[0])
 
    ##########
-   # Your code here
-   # Use parallel.prange() and a lock for each element of counts to parallelize
-   # data movement.  Be sure to avoid deadlock, and double-locking.
+   # Added requisite code below
    ##########
+   
    with nogil:
        for r in range(repeat):
-           for idx in range(src.shape[0]):
+           for idx in prange(src.shape[0], num_threads=4):
                if counts[src[idx]] > 0:
```

Comment:
should acquire lock before accessing counts! because counts could be changing in the meantime!
see what @sdaulton